File: calc_jre.py
from Get_global_value import J_type
from Get_global_value import num_q
from Get_global_value import Ez
import numpy as np


def calc_jre(AA, joints):
    n = len(joints)
    JJ_re = np.zeros((n, 3))
    if num_q == 0:
        print('Single body, there is no link')
    else:
        for i in range(n):
            A_I_i = AA[joints[i], :, :]
            if J_type[joints[i]] == 'R':
                JJ_re[i, :] = A_I_i @ Ez
            else:
                JJ_re[i, :] = np.zeros(3)

    return JJ_re

File: src/matlab/spacedyn_v2r1/calc_jre.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% The SpaceDyn, a MATLAB toolbox for Space and Mobile Robots.
% (C)1998 The Space Robotics Lab. directed by Kazuya Yoshida,
% Tohoku University, Japan.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	February 4, 1998, Last modification by K.Yoshida
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%
% CALC_JRE	Translational Jacobians w.r.t.
%           for the point specified by connection vector 'joints'
%
%   1998.1.12 A.Kurosu
%   2001.9.11 H.Hamano
%   2002.2.27 H.Hamano modified to version up
%

function JJ_re = calc_jre( LP, SV, joints )

global Ez


% Check number of joints
n = length(joints);


% If a Single body,
if LP.num_q == 0
   
   JJ_re = [];
   
% If a Multi body system,
else
   
   JJ_re = [];
   
   for i = 1 : 1 : n
      
%      A_I_i = SV.AA(:,joints(i)*3-2:joints(i)*3);
      
      % Rotational joint
      if LP.J_type(joints(i)) == 'R'
%         JJ_re = [ JJ_re A_I_i*Ez ];
         JJ_re = [ JJ_re SV.AA(:,joints(i)*3-2:joints(i)*3)*Ez ];
         
      % Prismatic joint
      else
         JJ_re = [ JJ_re [ 0 0 0 ]' ];
         
      end
      
   end
   
end


%%%EOF
